product-service/src/functions/get-products-by-id/handler.ts

```diff
@@ -1,28 +1,50 @@
 import 'source-map-support/register';
-
 import type {ValidatedEventAPIGatewayProxyEvent} from '@libs/apiGateway';
 import {formatJSONResponse} from '@libs/apiGateway';
 import {middyfy} from '@libs/lambda';
+import {Client} from 'pg';
+import schema from '@functions/get-products-by-id/schema';
 
-// @ts-ignore
-import products from '../../data/products-data.json';
+const {PG_HOST, PG_PORT, PG_DATABASE, PG_USERNAME, PG_PASSWORD} = process.env;
+const dbOptions = {
+    host: PG_HOST,
+    port: PG_PORT,
+    database: PG_DATABASE,
+    user: PG_USERNAME,
+    password: PG_PASSWORD,
+    ssl: {
+        rejectUnauthorized: false
+    },
+    connectionTimeoutMillis: 1000
+};
 
-// @ts-ignore
 const getProductsById: ValidatedEventAPIGatewayProxyEvent<typeof schema> = async (event) => {
+    console.log(event);
     try {
         const {productId = ''} = event.pathParameters;
-        const product = products.find(product => product.id === productId);
-        if (!product) {
+        const client = new Client(dbOptions);
+        await client.connect();
+        const text = 'select p.*, s.count from products p inner join stocks s on s.product_id = p.id where p.id = $1';
+        const values = [productId];
+        const {rows: products} = await client.query(text, values);
+        await client.end();
+        if (products.length === 0) {
             throw new Error('Product not found');
         }
+
+        const result = products.pop();
```

Comment:
I suggest you get array item by its index here - pop() seems too complicated for just getting array member.